src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java

```diff
@@ -480,23 +480,82 @@ private TypeBindings _bindingsForSubtype(JavaType baseType, int typeParamCount,
+            // `subclass`'s definition relative to the `baseType`.
+            // E.g. {@code subclass : Bar<B, X>} where {@code Bar<T, U> extends Foo<A<T>, U>}, would
+            // inherit type bindings B and X.  It would _not_ inherit `baseType`'s type binding to A[B]
+            // -- it would only inherit the inner type binding to B.
+            // This is because {@code Bar<?, ?>} already _is a_ {@code Foo<A<?>, ?>}, and serialization config
+            // for the class itself already comprehends the types that the inheritance specification
+            // has structurally locked into the base type's bindings.
+            JavaType[] bindings = new JavaType[baseCount];
             for (int i = 0; i < baseCount; ++i) {
-                types.add(baseType.containedType(i));
+                bindings[i] = _bindingForSubtypeAtBindingPosition(baseType, i, subclass);
             }
-            return TypeBindings.create(subclass, types);
+            return TypeBindings.create(subclass, bindings);
         }
         // Otherwise, two choices: match N first, or empty. Do latter, for now
         return TypeBindings.emptyBindings();
     }
 
+    private JavaType _bindingForSubtypeAtBindingPosition(JavaType baseType, int bindingPosition, Class<?> subclass) {
+        JavaType binding = baseType.containedType(bindingPosition);
+        // While the type binding is nested with exactly one inner binding, AND the subtype's definition
+        // itself already implies the outermost binding, then...
+        while (binding.containedTypeCount() == 1 && _subtypeImpliesBinding(baseType, binding, bindingPosition, subclass)) {
+            // ...peel the outer binding off
+            binding = binding.containedType(0);
+        }
+        return binding;
+    }
+
+    /**
+     * True if the subclass already implies the base type's binding at {@code bindingPosition}
+     * (see {@link JavaType#containedType}).  An example of an "implied" binding would be where
+     * the subclass is the erased class `C` (defined as {@code C<T> extends A<B<T>>}).  In this scenario, `A`
+     * would be the baseType and would be bound to `B` with a _nested_ inner binding to some type
+     * for `T`.  `C` structurally implies the `B` binding without it being present as a type parameter --
+     * it's part of the class definition, itself.
+     */
+    private boolean _subtypeImpliesBinding(JavaType baseType, JavaType baseTypeBinding, int bindingPosition, Class<?> subclass) {
+        // Should take a A<B<C>, D>, e.g., and make it A<B<?>, D> when bindingPosition == 0, e.g.
+        JavaType wildcardedBase = _fromClass(
+            null,
+            baseType.getRawClass(),
+            _wildcardedBindings(baseType, baseTypeBinding, bindingPosition)
+        );
+
+        JavaType[] wildcards = new JavaType[subclass.getTypeParameters().length];
+        Arrays.fill(wildcards, CORE_TYPE_OBJECT);
+        JavaType wildcardedSubclass = _fromClass(
```

Comment:
Maybe `wildcardedSubclass` isn't the best name for this.  Basically, it's `JavaType` representation of `subclass` that's equivalently fully-erased.